logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/PersistRegisteredClientIdUseCase.kt

```diff
@@ -0,0 +1,48 @@
+package com.wire.kalium.logic.feature.client
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.client.Client
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.conversation.ClientId
+import com.wire.kalium.logic.functional.fold
+
+/**
+ * Checks if the given ClientId is currently registered and persists it in the local storage if so, otherwise returns failure.
+ */
+interface PersistRegisteredClientIdUseCase {
+
+    /**
+     * @param clientId client id to be persisted
+     * @return success if the given id is registered, failure otherwise
+     */
+    suspend operator fun invoke(clientId: ClientId): PersistRegisteredClientIdResult
+}
+
+class PersistRegisteredClientIdUseCaseImpl(
```

Comment:
We still don't have a detekt rule for this, but it's a good idea to make this class internal (or its constructor), so we just communicate with clients using public interfaces :D